acme/acme/messages.py

```diff
@@ -285,7 +291,8 @@ class ExternalAccountBinding:
     """ACME External Account Binding"""
 
     @classmethod
-    def from_data(cls, account_public_key, kid, hmac_key, directory):
+    def from_data(cls, account_public_key: jose.JWK, kid: str, hmac_key: str,
+                  directory: Directory) -> 'ExternalAccountBinding':
```

Comment:
Well mypy is not complaining, because as of now `josepy` does not expose any types, so mypy assumes that `jws.JWS.sign` returns `Any`.

But this is wrong and you uncovered a potential inconsistency here. The caller of this code (`perform_registration` in `certbot._internal.client`) assumes that `ExternalAccountBinding.from_data` returns an instance of `ExternalAccountBinding`. This type annotation should be fixed as well.

However all of this means that the `NewRegistration` created by `perform_registration` actually holds a `Dict` in its `external_account_binding` field instead of an `ExternalAccountBinding` object. Even if in the end, the JSON representation sent to the ACME server is correct anyway, I wonder if it was really the intended purpose to use the `Dict` instead of the `ExternalAccountBinding` object.

cc @bmw

Reply:
I created https://github.com/certbot/certbot/issues/9044 with my thoughts and to track changing this.